qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/QuteDiagnostics.java

```diff
@@ -368,7 +397,16 @@ private static void validateSectionTag(Section section, Template template,
 	private static boolean canChangeContext(Section section) {
 		SectionKind sectionKind = section.getSectionKind();
 		return sectionKind == SectionKind.EACH || sectionKind == SectionKind.FOR || sectionKind == SectionKind.LET
-				|| sectionKind == SectionKind.SET || sectionKind == SectionKind.WITH;
+				|| sectionKind == SectionKind.SET || sectionKind == SectionKind.WITH
+				|| sectionKind == SectionKind.SWITCH || sectionKind == SectionKind.WHEN;
+	}
+
+	private static boolean isSwitchWhenSection(Section section) {

```

Comment:
The name `isSwitchWhenSection` can be confused, I think we should consider that we use switch and case for all code. I will rename `isSwitchWhenSection `to `isSwitchSection`